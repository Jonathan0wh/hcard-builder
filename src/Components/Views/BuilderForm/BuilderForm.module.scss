@import 'styles/colors';

.form {
  :global {
    h2.ant-typography {
      font-size: 28px;
      font-weight: 800;
      color: $bluewood;
      margin-bottom: 0;
    }
    h5 {
      font-size: 10px;
      font-weight: 700;
      color: $subtitle-grey;
      margin-top: 36px;
    }
    .ant-divider,
    .ant-divider-horizontal {
      margin: 0 0 25px 0;
    }
    .ant-btn {
      font-size: 20px;
      font-weight: 400;
      border: none;
      width: 190px;
      height: 36px;
    }
  }

  padding: 40px;
}

.button_col {
  text-align: center;
  margin-top: 30px;
}

.button_upload {
  background-color: $button-bluewood;
  color: $white;
  box-shadow: 0 2px $button-bluewood-shadow;

  label {
    font-size: 20px;
    font-weight: 400;
    color: $white;
    cursor: pointer;
  }

  input {
    display: none !important;
  }
}

.button_upload:hover,
.button_upload:focus {
  background-color: $button-bluewood-shadow;
  color: $white;
}

.button_create {
  background-color: $button-blue;
  color: $white;
  box-shadow: 0 2px $button-blue-shadow;
}

.button_create:hover,
.button_create:focus {
  background-color: $button-blue-shadow;
  color: $white;
}
