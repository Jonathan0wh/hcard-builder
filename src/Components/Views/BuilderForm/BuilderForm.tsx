import React from 'react';
import { Form, Typography, Row, Col, Divider, Button, Input } from 'antd';
import v from 'voca';

import { IFormData, IFormHandler } from 'Components/Views/HCardBuilder';
import FormField from 'Components/Shared/FormField';

import { STRINGS_FORM } from 'configs/strings';
import styles from './BuilderForm.module.scss';

interface BuilderFormProps {
  formData: IFormData;
  formHandler: IFormHandler;
}

const BuilderForm = ({ formData, formHandler }: BuilderFormProps) => {
  const { Title } = Typography;

  return (
    <Form layout="vertical" className={styles.form}>
      <Row>
        <Title level={2}>{STRINGS_FORM.FORM_TITLE}</Title>
      </Row>

      <h5>{STRINGS_FORM.FORM_SUBTITLE_PERSONAL_DETAILS}</h5>
      <Divider />
      <Row gutter={20}>
        <FormField
          label={STRINGS_FORM.FORM_LABEL_GIVEN_NAME}
          value={formData.givenName}
          setValue={formHandler.setGivenName}
        />
        <FormField
          label={STRINGS_FORM.FORM_LABEL_SURNAME}
          value={formData.surname}
          setValue={formHandler.setSurname}
        />
        <FormField
          label={STRINGS_FORM.FORM_LABEL_EMAIL}
          value={formData.email}
          setValue={formHandler.setEmail}
          type="email"
        />
        <FormField
          label={STRINGS_FORM.FORM_LABEL_PHONE}
          value={formData.tel}
          setValue={formHandler.setTel}
          type="tel"
        />
      </Row>

      <h5>{STRINGS_FORM.FORM_SUBTITLE_ADDRESS}</h5>
      <Divider />
      <Row gutter={20}>
        <FormField
          label={STRINGS_FORM.FORM_LABEL_HOUSE_NAME_OR_NUMBER}
          value={formData.house}
          setValue={formHandler.setHouse}
        />
        <FormField
          label={STRINGS_FORM.FORM_LABEL_STREET}
          value={formData.street}
          setValue={formHandler.setStreet}
        />
        <FormField
          label={STRINGS_FORM.FORM_LABEL_SUBURB}
          value={formData.suburb}
          setValue={formHandler.setSuburb}
        />
        <FormField
          label={STRINGS_FORM.FORM_LABEL_STATE}
          value={formData.state}
          setValue={formHandler.setState}
        />
        <FormField
          label={STRINGS_FORM.FORM_LABEL_POSTCODE}
          value={formData.postcode}
          setValue={formHandler.setPostcode}
        />
        <FormField
          label={STRINGS_FORM.FORM_LABEL_COUNTRY}
          value={formData.country}
          setValue={formHandler.setCountry}
        />
      </Row>

      <Row>
        <Col lg={12} className={styles.button_col}>
          <Button className={styles.button_upload}>
            <label
              htmlFor={v.kebabCase(STRINGS_FORM.FORM_BUTTON_UPLOAD_AVATAR)}
            >
              {STRINGS_FORM.FORM_BUTTON_UPLOAD_AVATAR}
            </label>
            <Input
              id={v.kebabCase(STRINGS_FORM.FORM_BUTTON_UPLOAD_AVATAR)}
              name={v.kebabCase(STRINGS_FORM.FORM_BUTTON_UPLOAD_AVATAR)}
              type="file"
              accept="image/*"
              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                e.target.files && formHandler.setAvatar(e.target.files[0])
              }
              className={styles.button_upload_input}
            />
          </Button>
        </Col>
        <Col lg={12} className={styles.button_col}>
          <Button className={styles.button_create}>
            {STRINGS_FORM.FORM_BUTTON_CREATE_HCARD}
          </Button>
        </Col>
      </Row>
    </Form>
  );
};

export default BuilderForm;
