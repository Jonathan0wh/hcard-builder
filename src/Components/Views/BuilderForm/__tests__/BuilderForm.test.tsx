import React from 'react';
import { shallow } from 'enzyme';

import BuilderForm from '../BuilderForm';

describe('<BuilderForm />', () => {
  const givenName = 'Sam',
    surname = 'Fairtax',
    email = 'sam.fairtax@fairtaxmedia.com.au',
    tel = '02 9282 2833',
    house = '1',
    street = 'Darling Island Road',
    suburb = 'Pyrmont',
    state = 'NSW',
    postcode = '2009',
    country = 'Australia',
    avatar = null;

  const formData = {
    givenName,
    surname,
    email,
    tel,
    house,
    street,
    suburb,
    state,
    postcode,
    country,
    avatar
  };

  const formHandler = {
    setGivenName: jest.fn(),
    setSurname: jest.fn(),
    setEmail: jest.fn(),
    setTel: jest.fn(),
    setHouse: jest.fn(),
    setStreet: jest.fn(),
    setSuburb: jest.fn(),
    setState: jest.fn(),
    setPostcode: jest.fn(),
    setCountry: jest.fn(),
    setAvatar: jest.fn()
  };

  it('should render by default', () => {
    const wrapper = shallow(
      <BuilderForm formData={formData} formHandler={formHandler} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
