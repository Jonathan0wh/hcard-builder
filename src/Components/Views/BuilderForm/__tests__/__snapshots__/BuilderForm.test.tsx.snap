// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuilderForm /> should render by default 1`] = `
<Form
  className="form"
  colon={true}
  hideRequiredMark={false}
  layout="vertical"
  onSubmit={[Function]}
>
  <Row
    gutter={0}
  >
    <Title
      level={2}
    >
      hCard Builder
    </Title>
  </Row>
  <h5>
    PERSONAL DETAILS
  </h5>
  <Divider />
  <Row
    gutter={20}
  >
    <FormField
      label="GIVEN NAME"
      setValue={[MockFunction]}
      value="Sam"
    />
    <FormField
      label="SURNAME"
      setValue={[MockFunction]}
      value="Fairtax"
    />
    <FormField
      label="EMAIL"
      setValue={[MockFunction]}
      type="email"
      value="sam.fairtax@fairtaxmedia.com.au"
    />
    <FormField
      label="PHONE"
      setValue={[MockFunction]}
      type="tel"
      value="02 9282 2833"
    />
  </Row>
  <h5>
    ADDRESS
  </h5>
  <Divider />
  <Row
    gutter={20}
  >
    <FormField
      label="HOUSE NAME OR #"
      setValue={[MockFunction]}
      value="1"
    />
    <FormField
      label="STREET"
      setValue={[MockFunction]}
      value="Darling Island Road"
    />
    <FormField
      label="SUBURB"
      setValue={[MockFunction]}
      value="Pyrmont"
    />
    <FormField
      label="STATE"
      setValue={[MockFunction]}
      value="NSW"
    />
    <FormField
      label="POSTCODE"
      setValue={[MockFunction]}
      value="2009"
    />
    <FormField
      label="COUNTRY"
      setValue={[MockFunction]}
      value="Australia"
    />
  </Row>
  <Row
    gutter={0}
  >
    <Col
      className="button_col"
      lg={12}
    >
      <Button
        block={false}
        className="button_upload"
        ghost={false}
        htmlType="button"
        loading={false}
      >
        <label
          htmlFor="upload-avatar"
        >
          Upload Avatar
        </label>
        <Input
          accept="image/*"
          className="button_upload_input"
          id="upload-avatar"
          name="upload-avatar"
          onChange={[Function]}
          type="file"
        />
      </Button>
    </Col>
    <Col
      className="button_col"
      lg={12}
    >
      <Button
        block={false}
        className="button_create"
        ghost={false}
        htmlType="button"
        loading={false}
      >
        Create hCard
      </Button>
    </Col>
  </Row>
</Form>
`;
